File: iterableapi/src/main/java/com/iterable/iterableapi/auth/classes/AuthFailure.java
package com.iterable.iterableapi.auth.classes;

import com.iterable.iterableapi.auth.enums.AuthFailureReason;

/**
 * Represents an auth failure object.
 */
public class AuthFailure {

    /** userId or email of the signed-in user */
    public final String userKey;

    /** the authToken which caused the failure */
    public final String failedAuthToken;

    /** the timestamp of the failed request */
    public final long failedRequestTime;

    /** indicates a reason for failure */
    public final AuthFailureReason failureReason;

    public AuthFailure(String userKey,
                       String failedAuthToken,
                       long failedRequestTime,
                       AuthFailureReason failureReason) {
        this.userKey = userKey;
        this.failedAuthToken = failedAuthToken;
        this.failedRequestTime = failedRequestTime;
        this.failureReason = failureReason;
    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/AuthFailure.kt
package com.iterable.iterableapi

/**
 * Represents an auth failure object.
 */
class AuthFailure(
    /** userId or email of the signed-in user */
    val userKey: String,
    /** the authToken which caused the failure */
    val failedAuthToken: String,
    /** the timestamp of the failed request */
    val failedRequestTime: Long,
    /** indicates a reason for failure */
    val failureReason: AuthFailureReason
)